cypress/support/pages/search.ts

```diff
@@ -1,10 +1,22 @@
 import { NAVIGATION } from "../../elements/navigation";
-import { SEARCH_PAGE_SELECTORS } from "../../elements/search-page";
+import { SEARCH } from "../../elements/search-page";
 
 export function navigateAndSearch(typedText) {
   cy.get(NAVIGATION.searchIcon)
     .click()
-    .get(SEARCH_PAGE_SELECTORS.searchInput)
+    .get(SEARCH.searchInput)
     .type(typedText, { delay: 500 })
     .should("have.value", typedText);
 }
+
+export function waitForSearchedProducts(typedText) {
+  cy.intercept("POST", Cypress.env("API_URI"), (req) => {
+    if (
+      req.body.operationName === "ProductCollection" &&
+      req.body.variables.filter.search === typedText
+    )
+      req.alias = "SearchResult";
+    req.continue();
+  });
+  cy.wait("@SearchResult");
+}
```

Comment:
This can fail. First, we should intercept the request, then we should make an action that will trigger that request, and after that wait for an aliased request. In this case, everything is passing because it takes w while, when searching for products is sending the request